packages/vuetify/src/components/VDataTable/VDataTableHeaderDesktop.ts

```diff
@@ -16,76 +16,95 @@ export default mixins(header).extend({
+        : this.$createElement('span', [header.text])
+      )
 
-      if (header.value === 'data-table-select' && !this.singleSelect) {
-        children.push(this.genSelectAll())
-      } else {
-        children.push(this.$scopedSlots[header.value]
-          ? this.$scopedSlots[header.value]!({ header })
-          : this.$createElement('span', [header.text])
-        )
-
-        if (!this.disableSort && (header.sortable || !header.hasOwnProperty('sortable'))) {
-          listeners['click'] = () => this.$emit('sort', header.value)
-
-          const sortIndex = this.options.sortBy.findIndex(k => k === header.value)
-          const beingSorted = sortIndex >= 0
-          const isDesc = this.options.sortDesc[sortIndex]
-
-          classes.push('sortable')
-
-          if (beingSorted) {
-            classes.push('active')
-            classes.push(isDesc ? 'desc' : 'asc')
-
-            attrs['aria-sort'] = isDesc ? 'descending' : 'ascending'
-            attrs['aria-label'] += isDesc
-              ? this.$vuetify.lang.t('$vuetify.dataTable.ariaLabel.sortDescending')
-              : this.$vuetify.lang.t('$vuetify.dataTable.ariaLabel.sortAscending')
-          } else {
-            attrs['aria-label'] += this.$vuetify.lang.t('$vuetify.dataTable.ariaLabel.sortNone')
-          }
-
-          if (header.align === 'end') children.unshift(this.genSortIcon())
-          else children.push(this.genSortIcon())
-
-          if (this.options.multiSort && beingSorted) {
-            children.push(this.$createElement('span', { class: 'v-data-table-header__sort-badge' }, [String(sortIndex + 1)]))
-          }
+      if (!this.disableSort && (header.sortable || !header.hasOwnProperty('sortable'))) {
+        data.on['click'] = () => this.$emit('sort', header.value)
+
+        const sortIndex = this.options.sortBy.findIndex(k => k === header.value)
+        const beingSorted = sortIndex >= 0
+        const isDesc = this.options.sortDesc[sortIndex]
+
+        data.class.push('sortable')
+
+        const { ariaLabel, ariaSort } = this.getAria(beingSorted, isDesc)
+
+        data.attrs['aria-label'] += `${header.text ? ':' : ''} ${ariaLabel}`
+        data.attrs['aria-sort'] = ariaSort
+
+        if (beingSorted) {
+          data.class.push('active')
+          data.class.push(isDesc ? 'desc' : 'asc')
         }
 
-        if (this.showGroupBy) {
-          children.push(this.genGroupByToggle(header))
+        if (header.align === 'end') children.unshift(this.genSortIcon())
+        else children.push(this.genSortIcon())
+
+        if (this.options.multiSort && beingSorted) {
+          children.push(this.$createElement('span', { class: 'v-data-table-header__sort-badge' }, [String(sortIndex + 1)]))
         }
       }
 
-      return this.$createElement('th', {
-        attrs,
-        class: classes,
-        style: styles,
-        on: listeners,
-      }, children)
+      if (this.showGroupBy) {
+        children.push(this.genGroupByToggle(header))
```

Comment:
Should be on one line like https://github.com/vuetifyjs/vuetify/pull/9638/files#diff-68e6ca531cb03e1916cea6cdc06df029R95